enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,435 @@
+
+After a bundle has been deployed to the cluster by the PO manager, admins may need to perform manual steps for upgrading that bundle version.
+
+That process may require the following steps:
+
+- Find the current bundle version for the configured PO package name.
+- Verifying that the redhat-operators index image contains a newer, higher semver bundle than what's currently applied to the cluster.
+- Deleting the underlying bundle deployment resources to force the PO manager to select a new bundle.
+
+These steps may require explicit documentation to provide an escape hatch for admins when the PO manager rolls out a vulnerable, or broken bundle version.
+
+#### Sourcing Bundles
+
+```mermaid
+sequenceDiagram
+participant ClusterVersion
+participant PO Manager
+participant Sourcing Layer
+participant CatalogSource
+PO Manager->>ClusterVersion: Watches
+ClusterVersion->>PO Manager: Cluster Version Range
+PO Manager->>Sourcing Layer: Desired Package Name
+PO Manager->>Sourcing Layer: Cluster Version
+Sourcing Layer->>CatalogSource: ListBundles
+CatalogSource->>Sourcing Layer: All Available Bundles
+Sourcing Layer-->Sourcing Layer: Filter By Package Name
+Sourcing Layer-->Sourcing Layer: Filter By Cluster Version Range
+Sourcing Layer-->Sourcing Layer: Filter By Highest Semver Package
+Sourcing Layer->>PO Manager: Selected registry+v1 Bundle
+```
+
+This phase will continue to use OLM's existing CatalogSource API in order to consume catalog content. In order to satisfy the admin intent for a specific package, the PO manager's sourcing layer will need to get the list of available bundles from the redhat-operator's CatalogSource resource, and then perform any client-side filtering logic to determine the "best bundle" to persist to the cluster. This process can be done using the ListBundles gRPC registry API, which will return a stream of bundles that are present in that catalog's index image.
+
+An initial implementation for the "best bundle" selection criteria may include selecting the highest semver bundle that's within the cluster version range requirements. Determining the cluster version information can be done by using a combination of the Kubernetes' discovery API, and the ClusterVersion resource. Future phases may aim to improve this bundle selection implementation through constructing upgrade graphs, evaluating package/cluster constraints, and performing dependency resolution.
+
+Once the list of available bundles within a catalog has been returned, additional filtering logic may be performed. As a result, a single registry+v1 bundle will be returned, and that bundle's metadata will be passed to the PO manager's application layer for further processing.
+
+#### Applying Bundles
+
+```mermaid
+sequenceDiagram
+participant PO Manager
+participant Application Layer
```

Comment:
not clear to me what "application layer" is here

Reply:
This can probably use some wordsmithing, but I think what I had in mind is that the POM will have initial abstractions for sourcing and apply bundle contents:

- Sourcing bundles via the current CatalogSource/gRPC registry APIs during phase 0, but we can pivot to other APIs entirely during subsequent phases (e.g. deppy APIs).
- Apply bundles via rukpak's BundleDeployment API during phase 0, but we can pivot to using a higher-level API during subsequent phases (e.g. a deppy "activator" component).

I'm open to suggestions on how to make that clearer though.